ca/ca.go

```diff
@@ -764,8 +779,18 @@ func (ca *CertificateAuthorityImpl) integrateOrphan() error {
 	if err != nil {
 		return fmt.Errorf("failed to parse orphan: %s", err)
 	}
-	issued := cert.NotBefore.Add(-ca.backdate)
-	_, err = ca.sa.AddCertificate(context.Background(), orphan.DER, orphan.RegID, orphan.OCSPResp, &issued)
+	issued := cert.NotBefore.Add(ca.backdate)
+	if orphan.Precert {
+		issuedNanos := issued.UnixNano()
+		_, err = ca.sa.AddPrecertificate(context.Background(), &sapb.AddCertificateRequest{
+			Der:    orphan.DER,
+			RegID:  &orphan.RegID,
+			Ocsp:   orphan.OCSPResp,
+			Issued: &issuedNanos,
+		})
+	} else {
+		_, err = ca.sa.AddCertificate(context.Background(), orphan.DER, orphan.RegID, orphan.OCSPResp, &issued)
```

Comment:
Should the last parameter be `&issuedNanos` similar to `AddPrecertificate`?

Reply:
I think AddCertificate does conversion in the gRPC wrapper.

Reply:
Yep, exactly.